src/independent_mogp.jl

```diff
@@ -115,10 +115,79 @@ Posterior implementation for isotopic inputs, given diagonal Σy (OILMM).
 See AbstractGPs.jl API docs.
 """
 function AbstractGPs.posterior(
-    ft::IsotropicFiniteIndependentMOGP, y::AbstractVector{<:Real}
+    ft::IsotropicByOutputsFiniteIndependentMOGP, y::AbstractVector{<:Real}
 )
     finiteGPs = finite_gps(ft, ft.Σy[1])
     ys = collect(eachcol(reshape(y, (length(ft.x.x), :))))
     ind_posts = [AbstractGPs.posterior(fx, y_i) for (fx, y_i) in zip(finiteGPs, ys)]
     return independent_mogp(ind_posts)
 end
+
+# AbstractGPs APIs implementations for MOInputIsotopicByFeatures.
+
+const IsotropicByFeaturesFiniteIndependentMOGP = FiniteGP{
+    <:IndependentMOGP,<:MOInputIsotopicByFeatures,<:Diagonal{<:Real,<:Fill}
+}
+
+function reorder_features_to_outputs_indices(x::MOInputIsotopicByOutputs)
```

Comment:
Maybe I don't fully understand your naming here but should this be changed to this:
```suggestion
function reorder_outputs_to_features_indices(x::MOInputIsotopicByOutputs)
```

Reply:
Or even just two `reorder_indices` dispatches?

Reply:
The message I'm trying to convey is "this function produces indices which are used to reorder things ordered by features into things ordered by outputs" -- I don't _think_ that they can also be used to go the other way, so I think it's probably important to make the distinction.

With that in mind, what name do you think might be better? I agree that the current one is a bit ambiguous.

I should probably also add a comment saying this...

Reply:
Hmm I need to see what this is actually doing then, since it must be different from what I was thinking it was doing. My initial confusion was how do we reorder something by features to by outputs if it is already a `MOInputIsotopicByOutputs`, but this must be me making assumptions as to what the function is doing

Reply:
From my understanding and after playing with the functions, `reorder_outputs_to_features_indices(x::MOInputIsotopicByOutputs)` returns the indices that would order it as a `MOInputIsotopicByFeatures` and `reorder_outputs_to_features_indices(x::MOInputIsotopicByFeatures)` returns the indices that would order it as a `MOInputIsotopicByOutputs`. Is that not the case?

Reply:
@thomasgudjonwright sorry for the amazing delay here. I've finally gotten around to revisiting this. I think you're right, and have left comments to that effect. Does this resolve this?

Reply:
Equally sorry for the delay! The to-do pile can get large pushing some things back eh 😄 These comments definitely clarify it for me. I guess I still find it slightly counterintuitive that something that **reorders by-output indices to by-feature indices** would be a function called `reorder_features_to_outputs_indices` since it sounds like its suggest a change from features to outputs. Not sure if that makes any sense

Reply:
Resolving given this is fairly outdated and has been addressed elsewhere